subsys/net/lib/openthread/platform/alarm.c

```diff
@@ -53,6 +53,12 @@ void platformAlarmProcess(otInstance *aInstance)
 	if (timer_ms_fired) {
 		timer_ms_fired = false;
 		otPlatAlarmMilliFired(aInstance);
```

Comment:
Is it ok that the `otPlat*` function will be called twice if the diag is enabled? The bare metal implementeation does it slightly different:
https://github.com/openthread/ot-nrf528xx/blob/main/src/src/alarm.c#L500
Note the `else` condition.

Reply:
You are right let's keep it similar.